libraries/c_sdk/standard/mqtt/README.md

```diff
@@ -1,18 +1,46 @@
+**Note:** The config `IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES` to override serializer APIs is not supported in the Compatibility Layer. [MQTT over BLE](https://docs.aws.amazon.com/freertos/latest/userguide/ble-demo.html#ble-demo-mqtt) was dependent on this config. However, starting from [202011.00 release](https://github.com/aws/amazon-freertos/releases/tag/202011.00), MQTT over BLE is redesigned to avoid the dependency on this config by using a [BLE transport implementation](../ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c). Please note that applications using MQTT over BLE with MQTT V2.x.x APIs will continue to work with this Compatibility Layer without having a dependency on the config `IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES`.
+
+## Comparison of code sizes of the Compatibility Layer and the coreMQTT library
+
+Code sizes are calculated on [202011.00 release](https://github.com/aws/amazon-freertos/releases/tag/202011.00). Please note that only the unique dependencies of these libraries are accounted for in this comparison and common dependencies such as the FreeRTOS Kernel, are not considered. Hence, the comparison will only show the added cost to the memory when choosing one of these libraries for developing an MQTT application. Refer to the tables below for the comparison of code sizes of the **Compatibility Layer** and **coreMQTT library** with their unique dependencies.
+
+**Code sizes for Compatibility Layer and dependencies**
+| Modules | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| Compatibility Layer | 15K | 13.5K |
+| coreMQTT | 6.9K | 5.7K |
+| Task Pool | 3.8K | 3.0K |
+| Platform Abstraction | 2.4K | 2.3K |
+| **Total estimate** | **28.1K** | **24.5K** |
+
+**Code sizes for coreMQTT library and dependencies**
+| Modules | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| coreMQTT | 6.9K | 5.7K |
+| Transport implementation | 3.1K | 2.9K |
+| **Total estimate** | **10K** | **8.6K** |
+
+**Note** Refer to the [CODESIZE.md](CODESIZE.md) for more details about the calculation of code sizes for the Compatibility Layer and the [coreMQTT FreeRTOS documentation](https://freertos.org/mqtt/index.html) for coreMQTT library.
+
+Note that coreMQTT library requires ~66% less code space as compared to the Compatibility Layer. **Hence, for applications requiring low memory footprint, we recommend using [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md), which doesn't have any dependency other than the standard C library, a customer-implemented network transport interface, and *optionally* a user-implemented platform time function.**
```

Comment:
Users will be confused by this statement. A compat layer is to provide backward compatibility - why would the user compare memry usage of the compat with coreMQTT. For the user, a comparison between old MQTT and coreMQTT makes more sense.

Reply:
This statement was to provide a recommendation to users of the Compatibility layer to consider using coreMQTT for applications requiring low memory foot print based on the memory numbers. The old MQTT library is not present in the amazon-freertos code but only the Compatibility layer.